interface/resources/qml/AssetServer.qml

```diff
@@ -164,6 +164,27 @@ Window {
         Window.copyToClipboard(url);
     }
 
+    function renameEl(index, data) {
+        if (!index) {
+            return false;
+        }
+
+        var path = assetProxyModel.data(index, 0x100);
+        if (!path) {
+            return false;
+        }
+
+        var destinationPath = path.split('/');
+        destinationPath[destinationPath.length - (path[path.length - 1] === '/' ? 2 : 1)] = data;
+        destinationPath = destinationPath.join('/').trim();
+
+        if (path === destinationPath) {
+            return;
+        }
+        if (!fileExists(destinationPath)) {
+            doRenameFile(path, destinationPath);
+        }
```

Comment:
Yes, the delay is a bit annoying.  But I don't want to make it look like it worked if it doesn't. 
But unless you have a big RTT to the server, it should be too fast to notice.